src/selection.rs

```diff
@@ -285,6 +285,9 @@ impl Selection {
     /// assert_eq!(matches[1], 2);
     /// ```
     pub fn list(&mut self, frame: &Frame) -> Result<Vec<usize>, Error> {
+        if self.size() != 1 {
+            panic!("can not call `Selection::list` on a multiple selection");
```

Comment:
One could return an error instead of panic. Do we have "a rule of thumb" in which cases a function panics or returns an error?

Reply:
I went for a panic because the function is documented to panic, but returning an error might be better in this case.

Regarding rule of thumb, panics should be for unrecoverable errors and programming error, and std::error::Error for recoverable errors.

Reply:
Updated to use a Result instead =)